framework/include/kernels/KernelBase.h

```diff
@@ -73,6 +73,12 @@ class KernelBase :
   /// Computes d-residual / d-jvar... storing the result in Ke.
   virtual void computeOffDiagJacobian(unsigned int jvar) = 0;
 
+  /// Compute this Kernel's contribution to the diagonal Jacobian entries
+  virtual void computeNonlocalJacobian() = 0;
```

Comment:
Nothing specific, just wanted to keep the design same as other jacobian methods such as `computeOffDiagJacobianScalar`. Probably not necessary.